src/main/kotlin/com/atkinsondev/opentelemetry/build/OpenTelemetryTestListener.kt

```diff
@@ -78,20 +77,44 @@ class OpenTelemetryTestListener(
             testTaskSpan.addEvent(TEST_FAILURE_SPAN_EVENT_NAME_KEY, attributesBuilder.build())
         }
 
-        val span = testSpanMap[testKey]
+        val span = testSpanMap.remove(test) ?: return
```

Comment:
I personally like early returns but I agree that it *does* make control flow slightly less obvious. Should I undo the change here?

Reply:
Good question, I'm also all for early returns - but the `return` on the right side of an `?:` does read a little odd to me.

What about making the null check a bit more obvious / verbose? Something like:

```
if (span == null) {
  return
}
```

Reply:
At least Jetbrains seems to consider `?: return` more idiomatic as IDEA sugguests using elvis when doing a `if (... == null) { return }`.

I'll just leave it as it was before 😃